src/bundler.ts

```diff
@@ -0,0 +1,28 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+
+import * as vscode from "vscode";
+
+const asyncExec = promisify(exec);
+
+export async function isGemOutdated(): Promise<boolean> {
+  try {
+    await execInPath("bundle outdated ruby-lsp");
+  } catch {
+    return true;
+  }
+
+  return false;
+}
+
+export async function updateGem(): Promise<void> {
+  await execInPath("bundle update ruby-lsp --conservative");
```

Comment:
I know this is not related to this PR, but can you please remove the conservative flag? I realized that if we updated the syntax_tree requirement, using conservative actually fails. Now that we're no longer putting the update window in people's faces, I think it's safe to use a regular update.